director/authentication.go

```diff
@@ -0,0 +1,40 @@
+package director
+
+import (
+	"github.com/gin-gonic/gin"
+	"github.com/pelicanplatform/pelican/config"
+	"github.com/pelicanplatform/pelican/param"
+	log "github.com/sirupsen/logrus"
+)
+
+type DiscoveryResponse struct {
+	JwksUri string `json:"jwks_uri"`
+}
+
+// Returns links related to director's authentication, including the link
+// to get the public key from the director
+func discoveryHandler(ctx *gin.Context) {
+	director := param.Federation_DirectorUrl.GetString()
+	rs := DiscoveryResponse{
+		JwksUri: director + "/.well-known/public_signing_key.jwks",
+	}
+
+	ctx.JSON(200, rs)
+}
+
+// Returns director's public key
+func jwksHandler(ctx *gin.Context) {
+	issuerKeyfile := param.IssuerKey.GetString()
+	key, err := config.LoadPublicKey("", issuerKeyfile)
+	if err != nil {
+		log.Errorf("Failed to load director's public key: %v", err)
+		ctx.JSON(500, gin.H{"error": "Failed to load director's public key"})
+	} else {
+		ctx.JSON(200, key)
```

Comment:
This is another spot where the JSON we return doesn't end with a newline character. Not sure if it's needed, but it's probably not a bad thing to always have.